Server/src/models/SelectQueryDatabase.ts

```diff
@@ -0,0 +1,2806 @@
+        .andWhere('subcategory.id = :subcategoryId', { subcategoryId: subcategory })
+        .getRawMany();
+    console.log(subcategoryPublicationData);
+
+    console.log("Getting authors");
+    let subcategoryAuthorData: IAuthorModel[] = await selectAuthorsQuery(connection.manager)
+        .innerJoin(Category, 'category', 'dataset.categoryId = category.id')
+        .innerJoin(Subcategory, 'subcategory', 'dataset.subcategoryId = subcategory.id')
+        .where('category.id = :categoryId', { categoryId: category })
+        .andWhere('subcategory.id = :subcategoryId', { subcategoryId: subcategory })
+        .getRawMany();
+    console.log(subcategoryAuthorData);
+
+    console.log("Getting data sets");
+    let subcategoryDatasetData = await selectDatasetsQuery(connection.manager)
+        .where('category.id = :categoryId', { categoryId: category })
+        .andWhere('subcategory.id = :subcategoryId', { subcategoryId: subcategory })
+        .getRawMany();
+    console.log(subcategoryDatasetData);
+
+    console.log("Getting data set materials");
+    let subcategoryMaterialData = await selectMaterialQuery(connection.manager)
+        .innerJoin(Category, 'category', 'dataset.categoryId = category.id')
+        .innerJoin(Subcategory, 'subcategory', 'dataset.subcategoryId = subcategory.id')
+        .where('category.id = :categoryId', { categoryId: category })
+        .andWhere('subcategory.id = :subcategoryId', { subcategoryId: subcategory })
+        .getRawMany();
+    console.log(subcategoryMaterialData);
+
+    console.log("Getting data point data");
+    let subcategoryDatapointData = await selectDataPointsQuery(connection.manager)
+        .innerJoin(Category, 'category', 'dataset.categoryId = category.id')
+        .innerJoin(Subcategory, 'subcategory', 'dataset.subcategoryId = subcategory.id')
+        .where('category.id = :categoryId', { categoryId: category })
+        .andWhere('subcategory.id = :subcategoryId', { subcategoryId: subcategory })
+        .getRawMany();
+    console.log(subcategoryDatapointData);
+
+    console.log("Getting data point comments");
+    let subcategoryDatapointComments = await selectDataPointCommentsQuery(connection.manager)
+        .innerJoin(Category, 'category', 'dataset.categoryId = category.id')
+        .innerJoin(Subcategory, 'subcategory', 'dataset.subcategoryId = subcategory.id')
+        .where('category.id = :categoryId', { categoryId: category })
+        .andWhere('subcategory.id = :subcategoryId', { subcategoryId: subcategory })
+        .getRawMany();
+    console.log(subcategoryDatapointComments);
+
+    const allData: IDatasetResponseModel = {
+        authors: subcategoryAuthorData,
+        publications: subcategoryPublicationData,
+        dataPoints: subcategoryDatapointData,
+        dataset: subcategoryDatasetData,
+        materials: subcategoryMaterialData,
+        dataPointComments: subcategoryDatapointComments
```

Comment:
There is tons of good queries here but working through this is impossible. We should group common queries together